editor/editor_properties.cpp

```diff
@@ -462,10 +462,50 @@ void EditorPropertyPath::_set_read_only(bool p_read_only) {
 }
 
 void EditorPropertyPath::_path_selected(const String &p_path) {
-	emit_changed(get_edited_property(), p_path);
+	StringName edited_property = get_edited_property();
+	emit_changed(edited_property, p_path);
+	if (edited_property == "exec_path") {
+		emit_signal("exec_path_changed", p_path);
+	}
 	update_property();
 }
 
+void EditorPropertyPath::_on_exec_path_changed(String new_path) {
+	Ref<RegEx> regex;
+	regex.instantiate();
+
+	const String editor_pattern = R"([\\/]((?:[jJ]et[Bb]rains\s*)?[rR]ider(?:\s*(EAP|\d{4}\.\d+|\d{4}\.\d+\s*[Dd]ev)?)?|[vV]isual\s*[sS]tudio\s*[cC]ode|[sS]ubl(ime\s*[tT]ext)?|sublime_text|(g)?vim|emacs|atom|geany|kate|code)(?:\.app|\.exe|\.bat|\.sh)?)";
```

Comment:
There's a way to make the regex case-insensitive, so you don't need to check for both uppercase and lowercase:

- Use the case-insensitive regex flag.
- Make the source text lowercase before matching.